docs/cpu-metrics-in-fleet.md

```diff
@@ -0,0 +1,374 @@
+      query.
+   3. Recall that the values stored in `system.process.cpu.total.value` field are counter values, meaning they are expected
+      to be non-decreasing over time. So to consider a single CPU utilization value for each 1-minute bucket, we take the
+      `max(system.process.cpu.total.value)`. This is the `max_cpu` aggregation in the Elasticsearch query.
+   4. Given that the values stored in `system.process.cpu.total.value` field are counter values, to get CPU utilization
+      (in milliseconds) for each 1-minute period, we subtract the `max(system.process.cpu.total.value)` of a 1-minute
+      bucket from the same value of the previous 1-minute bucket. Further, we recalculate this value for 10-second buckets
+      (essentially dividing the 1-minute bucket value by 6). This is all done by the `cpu_derivative` aggregation in the
+      Elasticsearch query.
+   5. this per-10-second CPU utilization (in milliseconds) is divided by `10 * 1000` (= 10 seconds expressed as milliseconds),
+      to get the _percentage_ CPU utilization over 10 elapsed seconds. This is the `cpu` aggregation in the Elasticsearch query.
+   6. So now we end up with five CPU utilization (in %) values, one for each minute.
+   7. These are averaged to arrive at the CPU utilization (in %) over five minutes. This is the `avg_cpu` aggregation in
+      the Elasticsearch query.
+2. The 5-minute average CPU utilization (in %) for each of the processes is summed up to arrive at a total CPU utilization
+   (in %) for the Agent process and all Beat processes managed by Agent. This is the `sum_cpu` aggregation in the Elasticsearch
+   query. The result is the total CPU utilization (in %) by Agent and it's child Beat processes across all available cores
+   (as opposed to being normalized per-core). For example, if a machine has 8 cores, the resulting value will be in the range of
+   (0%, 800%).
+
+## Observations
+
+* The CPU utilization (in %) of Agent and each of the Beat processes can vary wildly. Generally speaking, the Agent process
+  itself does not utilize much CPU. And each Beat process may utilize CPU depending on the type of computations it is performing
+  on the data.
+
+* Also, CPU utilization is rarely constant. If the output of `top` or `htop` for Agent and Beat processes is observed over
+  time, the CPU utilization % shown varies for each process over time.
+
+* To relate the output seen in `top` or `htop` for Agent and Beat processes with the single value shown in the Fleet UI,
+  one must observe the values in the `top` / `htop` output over five minutes, arrive at an average CPU utilization (in %)
+  value for each process, and sum it up.  The resulting value will roughly be equal to the single value shown for that Agent
+  in the Agent Listing page in the Fleet UI.
+
+## Suggested improvements
+
+* We should document the observations above in an appropriate location and perhaps link to this documentation from the "i"
```

Comment:
Ideally, the tooltip text should also be improved, as right now it's pretty vague.

Reply:
Yeah, if we can **briefly** explain what the metric represents and a bit about how it's calculated, we should put all that in the tooltip.  Otherwise, I think we should just replace the tooltip with a  link to a documentation page.

Reply:
IMO if we have to link to a documentation page we are probably presenting the wrong value. Most likely the CPU should just document itself as the 5 minute average CPU utilization, or switch to just presenting an instantaneous value from the last checkin with a link to see a graph of the values over time.

Reply:
Agreed, I guess this suggestion should be more of a fallback if we can't come up with a good UX in the Fleet UI itself.

> ... just presenting an instantaneous value from the last checkin 

I'm partial to idea of showing a single value that's the sum the of the latest CPU utilization %s for Agent + all it's component processes with a tooltip explaining that the value can range from 0 to the number of cores * 100.  This value should be easily correlatable to what's seen in the output of `top` / `htop`.

> ... with a link to see a graph of the values over time.

++,  single value could link to a chart showing the CPU utilization %, per process, over time, in case the user wants to drill down further.